apps/studio/src/server/modules/resource/resource.service.ts

```diff
@@ -362,25 +365,40 @@ export const getLocalisedSitemap = async (
         .leftJoin("Blob as published", "Version.blobId", "published.id")
         .select(() => [headerSql, thumbnailSql, ...defaultResourceSelect]),
     )
-    .with("childCousinIndexPages", (eb) => {
-      return eb
+    // Step 3: Get all nested folders and collections
+    .withRecursive("nestedResources", (eb) =>
+      eb
         .selectFrom("Resource")
-        .where("parentId", "in", (qb) =>
-          qb
-            .selectFrom("immediateSiblings")
-            .select("id")
-            .where("type", "in", [
-              ResourceType.Collection,
-              ResourceType.Folder,
-            ]),
-        )
-        .where("type", "=", ResourceType.IndexPage)
-        .where("state", "=", "Published")
-        .leftJoin("Version", "Version.id", "publishedVersionId")
+        .where("Resource.siteId", "=", siteId)
+        .where("Resource.type", "in", [
+          ResourceType.Folder,
+          ResourceType.Collection,
+          ResourceType.IndexPage,
+        ])
+        .where("Resource.state", "=", ResourceState.Published)
+        .leftJoin("Version", "Version.id", "Resource.publishedVersionId")
```

Comment:
[perf] can we try this on staging (or on local?) - i had a similar query and the performance wasn't great so i'm worried that this iwll negatively impact load times

Reply:
[stupid question] won't this result in more data than required for nested resources? eg, i'm previewing a page C at `A -> B -> C`, this will still get the list of all resources that are collections/folders/index pages?

Reply:
yes it will, and thats intentional (unfortunately)

this is because collection can be pretty nested down (e.g. root -> 1 or more folders -> collection). Without fetching all the way down, it will crash since studio cannot find the collection in the sitemap

Reply:
@seaerchin tried on staging (locally)
- for the root page of the largest site, its between 180ms to 480ms (median 300ms)
- for smaller sites (<100ms)

this is including network call (whats observed from the dev. console)

Reply:
i'm ok with approving this - 480ms seems reasonable as an upper bound, but we need to be clear on:
1. what is the expected upper bound of resources we expect to get (ie, for direct children, what's the highest amount)
2. what's our deepest level for resources (ie, how nested do we allow our resources to be?)

i'm ok with this because it shouldn't impact load times negatively **at this point**, but i'm not sure if we need to do more work wrt this query (out of scope for this)

cc @harishv7 or @dcshzj if yall have objections

Reply:
@harishv7 

using site 1 for staging, total resources: 5476 with max depth of 7
```sql
WITH RECURSIVE tree AS (
  SELECT
    id,
    "parentId",
    1 AS depth
  FROM "Resource"
  WHERE "parentId" IS NULL and "siteId" = 1 and "type" in ('Folder', 'Collection', 'IndexPage')

  UNION ALL

  SELECT
    i.id,
    i."parentId",
    t.depth + 1
  FROM "Resource" i
  JOIN tree t ON i."parentId" = t.id
  where "siteId" = 1
)
SELECT MAX(depth) AS max_depth FROM tree;
```